### PR TITLE
[BUGFIX] Don't use destroyApp if new API is being used

### LINKED ISSUE #16180: Generated test(s) referencing `destroyApp` fail out of the box using Beta
Several of the test blueprints have already been updated in https://github.com/emberjs/ember.js/issues/15933 in order to support the new testing API approved in [RFC 232](https://github.com/emberjs/rfcs/blob/master/text/0232-simplify-qunit-testing-api.md.).

Within a new beta app (3.0.0-beta.4) and generating an instance-initializer using the blueprint merged in https://github.com/emberjs/ember.js/pull/15945, the tests fail out of the box due to the `destroyApp` helper not being present within the newly generated app:

```sh
Error: Could not find module `<app_name>/tests/helpers/destroy-app` 
imported from `<app_name>/tests/unit/instance-initializers/foo-test.js`
```

Is this helper supposed to be part of a new app, or are the blueprints incorrect?

### PR DESCRIPTION
**Pull request by Serabe**
Fixes #16180

### CODE REVIEW

**Comment by rwjblue** [comment]
This seems good, but I think you also need to fixup the fixture that we test against (the blueprint tests are failing)...

**Comment by Serabe** [comment]
Will fix on Monday.

**Comment by mixonic** [comment]
Ping as a friendly reminder @Serabe!

**Comment by nag5000** [comment]
instance-initializer-test blueprint also needs to be updated.

**Comment by Serabe** [comment]
Had a tough week. Try to do it tomorrow, as today I am attending a conference.

**Comment by Serabe** [comment]
Found the problem. I forgot to commit two files (－‸ლ)

File: blueprints/initializer-test/qunit-files/tests/unit/initializers/__name__-test.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ module('<%= friendlyTestName %>', {
     });
   },
   afterEach() {
-    destroyApp(this.application);
+    <% if (destroyAppExists) { %>destroyApp(this.application);<% } else { %>run(application, 'destroy');<% } %>
```
Comment (nag5000) [inline blueprints/initializer-test/qunit-files/tests/unit/initializers/__name__-test.js]: should be `run(this.application, ...)`?

Comment (Serabe) [reply blueprints/initializer-test/qunit-files/tests/unit/initializers/__name__-test.js]: Done. Thank you

**Pull request merged**